aws-sdk-core/spec/aws/cloud_front/cookie_signer_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+module Aws
+  module CloudFront
+    describe CookieSigner do
+
+      let(:options) {
+        {
+          :key_pair_id => "CF_KEYPAIR_ID",
+          :private_key_path =>"#{File.dirname(__FILE__)}/cf_private_key.pem"
+        }
+      }
+      let(:signer) { Aws::CloudFront::CookieSigner.new(options) }
+      let(:expires) { 1357034400 } # January 1, 2013 10:00 am UTC (Unix timestamp)
+
+      describe '#signed_cookie' do
+        it 'raises error if url is invaild' do
+          expect {
+            signer.signed_cookie("what_ever_ilegal/url")
+          }.to raise_error(ArgumentError)
+        end
+        it 'can generate signed urls with custom policy' do
+          policy =  {
+            'Statement' => [
+              'Resource' => 'images/image.jpg',
+                'Condition' => {
+                'IpAddress' => {'AWS:SourceIp' => '10.52.176.0/24'},
+                'DateLessThan' => {'AWS:EpochTime' => expires}
+                }
+              ]
+          }
+          cookie = signer.signed_cookie(
+            "http://abc.cloudfront.net/images/image.jpg",
+            policy: policy.to_json
+          )
+          expected = {'CloudFront-Policy' => 'eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaW1hZ2VzL2ltYWdlLmpwZyIsIkNvbmRpdGlvbiI6eyJJcEFkZHJlc3MiOnsiQVdTOlNvdXJjZUlwIjoiMTAuNTIuMTc2LjAvMjQifSwiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjEzNTcwMzQ0MDB9fX1dfQ__',
```

Comment:
Using hash is fine, I'd prefer having `expect` key by key, which might make debugging easier in the future.

Perhaps something like 
``` ruby
expect(cookie['CloudFront-Policy']).to eq(' ... ')
expect(cookie['CloudFront-Signature']).to eq(' ... ')
expect(cookie['CloudFront-Key-Pair-Id']).to eq(' ... ')
```